### PR TITLE
Added a few refinements to make the repo more user friendly.

### PR DESCRIPTION
**Pull request by WadeWaldron**
## Description

* Enhanced the [README.md](README.md)
* Added the [cheatsheet.md](cheatsheet.md)
* Added the `build.sh`
* Added a PR Template.

## Checklist

- [x] Run all tests with `./build.sh validate`.
- [x] Update the [CHANGELOG.md](CHANGELOG.md).
- [x] Update the [README.md](README.md) if necessary.

### CODE REVIEW

**Review by afzal-confluent — APPROVED** [review]
Tiny nitpiks.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,77 @@
...
+
+You can automatically solve an exercise by running:
+
+```bash
+./exercise.sh solve <exercise number>
+```
+
+**Note:** Solving an exercise will overwrite your code.
+
+You can solve just single file by running:
```
[the remark this answers is not in the document]
Comment (WadeWaldron) [reply README.md:65]:
```suggestion
You can solve a single file by running:
```

File: README.md [hunk 2]
```diff
@@ -1,3 +1,77 @@
...
-This repository is for the **Building Flink Applications in Java** course provided by Confluent Developer.
\ No newline at end of file
+This repository is for the **Building Flink Applications in Java** course provided by Confluent Developer.
+
+[https://developer.confluent.io/courses/flink-java](https://developer.confluent.io/courses/flink-java)
+
+## Requirements
+
+- Java 11 
+	- Flink does not currently support anything newer than Java 11. And the code in the repo assumes that you are using Java 11.
```
[the remark this answers is not in the document]
Comment (WadeWaldron) [reply README.md:10]:
```suggestion
	- Flink does not currently support anything newer than Java 11. The code in the repo assumes that you are using Java 11.
```

File: README.md [hunk 3]
```diff
@@ -1,3 +1,77 @@
...
+- A Java development environment.
+- A Flink installation.
+- A Confluent Cloud account.
+
+## Gitpod
+
+A Gitpod configuration is available for these exercises. You can use this to construct a pre-configured environment suitable for working on the exercises:
```
[the remark this answers is not in the document]
Comment (WadeWaldron) [reply README.md:17]:
```suggestion
A [Gitpod](https://gitpod.io/) configuration is available for these exercises. You can use this to construct a pre-configured environment suitable for working on the exercises:
```

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+## Description
+
+Enter your description here.
+
+## Checklist
+
+- [ ] Run all tests with `./build.sh validate`.
```
Comment (WadeWaldron) [inline .github/pull_request_template.md:7]:
```suggestion
- [ ] Unit tests created/updated for any new code (where applicable).
- [ ] Run all tests with `./build.sh validate`.
```

**Pull request merged**